src/manifolds/Symplectic.jl

```diff
@@ -0,0 +1,889 @@
+@doc raw"""
+    Symplectic{n, ℝ} <: AbstractEmbeddedManifold{ℝ, DefaultIsometricEmbeddingType}
+
+Over the real number field ℝ the elements of the symplectic manifold
+are all $2n × 2n$ matrices satisfying the requirement
+````math
+\operatorname{Sp}(2n, ℝ) = \bigl\{ p ∈ ℝ^{2n × 2n} \, \big| \, p^TQ_{2n}p = Q_{2n} \bigr\},
+````
+where
+````math
+Q_{2n} =
+\begin{bmatrix}
+  0_n & I_n \\
+ -I_n & 0_n
+\end{bmatrix},
+````
+with $0_n$ and $I_n$ denoting the $n × n$ zero-matrix
+and indentity matrix in ``ℝ^{n \times n}`` respectively.
+
+Internally the dimensionality of the structure is stored as half of the even dimension
+supplied to the constructor, `Symplectic(2n) -> Symplectic{n}()`,
+as most computations with points on the Real Symplectic manifold takes advantage of the
+natural block structure
+of a matrix ``A ∈ ℝ^{2n × 2n}`` where we consider it as consisting of four
+smaller matrices in ``ℝ^{n × n}``.
+"""
+struct Symplectic{n,𝔽} <: AbstractEmbeddedManifold{𝔽,DefaultIsometricEmbeddingType} end
```

Comment:
With very little work we could extend it to the symplectic Lie group, what do you think @kellertuer ? This could be done similarly to `GeneralLinear`.

Reply:
I think it might be a little complicated right now and much more doable once the new interface is there.

Reply:
Yes, this should be easier in the new interface.